assembl/graphql/document.py

```diff
@@ -56,9 +60,14 @@ def mutate(root, args, context, info):
 
         uploaded_file = args.get('file')
         if uploaded_file is not None:
+
             # Because the server is on GNU/Linux, os.path.basename will only work
             # with path using "/". Using ntpath works for both Linux and Windows path
             filename = ntpath.basename(context.POST[uploaded_file].filename)
+            extension = filename.split('.')[~0]
```

Comment:
Such a fancy way to say `-1`

Reply:
Just a note, file extension is NOT reliable enough to determine the type of file. The `context.POST[uploaded_file].mime_type` isn't much better either unfortunately, and can differ depending on the client. You could make the check that the extension and the broader category match (jpg in `image/jpeg`) for example. But this will add enormous complexity to code, as you'll have to have fallbacks and put extensions into groups (browser determined mime-types are unreliable too).

Reply:
Yes, basicly file extension is not enough. But the web server try to guess the engine to use based on it, so if you send a php script call `image.jpg`, the webserver will try to render it as an image and will never call the php engine (if any).
I'll see if I can find a python lib for checking magic numbers.

Reply:
There is a lib called `filemagic` which does that, it checks the filetype based on the first bytes of a file.

Reply:
We use `python-magic` in https://github.com/assembl/url_metadata to detect the file type. See https://github.com/assembl/url_metadata/blob/8e3ed811a7a2ee0a54eac6bf40b9044b902a6443/url_metadata/__init__.py#L153

Reply:
Please use `-1` instead of `~0`.

Reply:
Done with `python-magic`.